### DISCUSSION TITLE
AR examples are slow and crash Safari IOS 14.4

### DISCUSSION THREAD

**Discussion by alew3** [google/model-viewer#2244] [Q&A]
### Description

When I try out the examples under https://modelviewer.dev/examples/augmentedreality/#ar using IOS 14.4 with an IPhone X, everything is slow and stutters until after 1-2s it crashes Safari.

I have tried before these examples on my Phone and it was running very smooth with no issues. Not sure if a model viewer update or Safari IOS might have broken things.

#### Live Demo
Here is a screen capture:
https://www.dropbox.com/s/row636vxsaoksiy/RPReplay_Final1614798171.mov?dl=0

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [X ] Safari

### OS

- [ ] Android
- [X ] iOS 14.4
- [ ] Linux
- [ ] MacOS
- [ ] Windows

**Comment by elalish** [comment]
Wow, that's ugly! Very strange since my ancient iPhone SE is also on iOS 14.4 and that same site is nice and smooth, despite a much older GPU. Can you file a bug on Safari?

**Comment by alew3** [comment]
> Wow, that's ugly! Very strange since my ancient iPhone SE is also on iOS 14.4 and that same site is nice and smooth, despite a much older GPU. Can you file a bug on Safari?

I did a simple model viewer demo and it still works very smoothly. Not sure if the site demo changed anything, but it worked fine previously.

**Comment by elalish** [comment]
@kenrussell This is the Safari regression I was talking about.

**Comment by alew3** [comment]
I tested on my IPad Pro 2018 iPadOS 14.4 and it works fine. Not sure if it is a specific problem with my phone.

**Comment by kenrussell** [comment]
Please surface this to WebKit's engineers by filing a bug with a test case on bugs.webkit.org and post the link to the bug here. Thanks.

**Comment by ryancrunchi** [comment]
Is your device refurbished ?
I have experienced weird behaviours on refurbished iPhones, even with native app development.
